ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUnixTime.java

```diff
@@ -99,7 +98,6 @@ public void configure(MapredContext context) {
     if (context != null) {
       String timeZoneStr = HiveConf.getVar(context.getJobConf(), HiveConf.ConfVars.HIVE_LOCAL_TIME_ZONE);
       timeZone = TimestampTZUtil.parseTimeZone(timeZoneStr);
-      FORMATTER.withZone(timeZone);
```

Comment:
is it dead only, or the folks did it wrong by not assigning it back to the FORMATTER?
```
      FORMATTER = FORMATTER.withZone(timeZone);
```
Earlier it was modifying the formatter by a setter method
```
  formatter.setTimeZone(TimeZone.getTimeZone(timeZone));
```

Reply:
The part of the code that does the translation from a long representing epoch seconds to string on a specific `timeZone` (line 121 to 125), does not need the formatter to have specific timezone since it uses `ZonedDateTime`.

Explicitly setting the formatter to the same timezone with `ZonedDateTime` does not alter the result as the following example illustrates: 

```java
    ZoneId singaporeZone = ZoneId.of("Asia/Singapore");
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss");

    Instant instant = Instant.now();
    ZonedDateTime zonedDT = ZonedDateTime.ofInstant(instant, singaporeZone);

    System.out.println(zonedDT.format(formatter));
    System.out.println(zonedDT.format(formatter.withZone(singaporeZone)));
```
It would make sense to set `FORMATTER` to a specific zone if it was different from the timezone in `ZonedDateTime`, but this is not the case here.

Reply:
hey, sorry I missed it